src/util/FullMath.ts

```diff
@@ -10,8 +11,32 @@ export abstract class FullMath {
   static mulDivRoundingUp(a: JSBI, b: JSBI, denominator: JSBI): JSBI {
     const product = JSBI.multiply(a, b);
     let result = JSBI.divide(product, denominator);
-    if (JSBI.notEqual(JSBI.remainder(product, denominator), ZERO))
+    if (JSBI.greaterThan(JSBI.remainder(product, denominator), ZERO)){
+      assert(JSBI.lessThan(result, MaxUint256), "OVERFLOW");
       result = JSBI.add(result, ONE);
+    }
     return result;
   }
+
+  
+  static isEqualApproximately(a: JSBI, b: JSBI, errorInMinimumUnit: JSBI) {
```

Comment:
Is this still needed if we have achieved absolute precision?
If so, consider changing it into something like `equalsWithTolerance(a, b, toleranceInMinUnit)`